test/jit/test_complex.py

```diff
@@ -173,18 +170,148 @@ def forward(self, infj: int, nanj: int):
+        self.checkScript(fn_float_bool, (-3.0, True, ))
+        self.checkScript(fn_float_bool, (6.0, False, ))
+
+        def fn_float_int(real: float, img: int):
+            return complex(real, img)
+
+        self.checkScript(fn_float_int, (0.0, 1, ))
+        self.checkScript(fn_float_int, (0.0, -1, ))
+        self.checkScript(fn_float_int, (1.8, -3, ))
+        self.checkScript(fn_float_int, (2.7, 8, ))
+
+        def fn_int_float(real: int, img: float):
+            return complex(real, img)
+
+        self.checkScript(fn_int_float, (1, 0.0, ))
+        self.checkScript(fn_int_float, (-1, 1.7, ))
+        self.checkScript(fn_int_float, (-3, 0.0, ))
+        self.checkScript(fn_int_float, (2, -8.9, ))
+
+    def test_torch_complex_constructor_with_tensor(self):
+        tensors = ([torch.rand(1), torch.randint(-5, 5, (1, )), torch.tensor([False])])
+
+        def fn_tensor_float(real, img: float):
+            return complex(real, img)
+
+        def fn_tensor_int(real, img: int):
+            return complex(real, img)
+
+        def fn_tensor_bool(real, img: bool):
+            return complex(real, img)
+
+        def fn_float_tensor(real: float, img):
+            return complex(real, img)
+
+        def fn_int_tensor(real: int, img):
+            return complex(real, img)
+
+        def fn_bool_tensor(real: bool, img):
+            return complex(real, img)
+
+        for tensor in tensors:
+            self.checkScript(fn_tensor_float, (tensor, 1.2))
+            self.checkScript(fn_tensor_int, (tensor, 3))
+            self.checkScript(fn_tensor_bool, (tensor, True))
+
+            self.checkScript(fn_float_tensor, (1.2, tensor))
+            self.checkScript(fn_int_tensor, (3, tensor))
+            self.checkScript(fn_bool_tensor, (True, tensor))
+
+        def fn_tensor_tensor(real, img):
+            return complex(real, img) + complex(2)
+
+        for x, y in product(tensors, tensors):
+            self.checkScript(fn_tensor_tensor, (x, y, ))
+
     def test_comparison_ops(self):
         def fn1(a: complex, b: complex):
             return a == b
 
         def fn2(a: complex, b: complex):
             return a != b
 
+        def fn3(a: complex, b: float):
+            return a == b
+
+        def fn4(a: complex, b: float):
+            return a != b
+
         x, y = 2 - 3j, 4j
         self.checkScript(fn1, (x, x))
         self.checkScript(fn1, (x, y))
         self.checkScript(fn2, (x, x))
-        self.checkScript(fn1, (x, y))
+        self.checkScript(fn2, (x, y))
+
+        x1, y1 = 1+0j, 1.0
+        self.checkScript(fn3, (x1, y1))
+        self.checkScript(fn4, (x1, y1))
+
+        def fn5(a: complex, b: int):
+            return a == b
+        def fn6(a: complex, b: int):
+            return a != b
+
+        with self.assertRaises(RuntimeError):
+            self.checkScript(fn5, (x1, 1))
+        with self.assertRaises(RuntimeError):
+            self.checkScript(fn6, (x1, 1))
```

Comment:
Can you use `assertRaisesRegex` or `assertRaisesRegexWithHighlight` here instead? That way we can test the error message too.